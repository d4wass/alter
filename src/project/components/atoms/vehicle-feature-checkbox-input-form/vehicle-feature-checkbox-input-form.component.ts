import { Component, Input } from '@angular/core';
import { ControlsOf, FormGroup } from '@ngneat/reactive-forms';

@Component({
  selector: 'app-vehicle-feature-checkbox-input-form',
  template: `
    <form [formGroup]="equipmentCtrl">
      <section class="example-section">
        <mat-checkbox formControlName="appleCarPlay" class="example-margin"
          >Apple Car Play</mat-checkbox
        >
        <mat-checkbox formControlName="androidAuto" class="example-margin"
          >Android Auto</mat-checkbox
        >
        <mat-checkbox formControlName="bluetooth" class="example-margin">Bluetooth</mat-checkbox>
        <mat-checkbox formControlName="usb" class="example-margin">USB</mat-checkbox>
        <mat-checkbox formControlName="headUpDisplay" class="example-margin"
          >headup display</mat-checkbox
        >
        <mat-checkbox formControlName="navigation" class="example-margin">navigation</mat-checkbox>
        <mat-checkbox formContolName="airConditioning" class="example-margin"
          >air conditioning</mat-checkbox
        >
        <mat-checkbox formControlName="keyLess" class="example-margin">keyless access</mat-checkbox>
        <mat-checkbox formControlName="isofix" class="example-margin">isofix</mat-checkbox>
        <section>
          <h5>Cruise Control</h5>
          <form formGroupName="cruiseControl">
            <mat-checkbox formControlName="standard" class="example-margin">standard</mat-checkbox>
            <mat-checkbox formControlName="active" class="example-margin">active</mat-checkbox>
            <mat-checkbox formControlName="adaptive" class="example-margin">adaptive</mat-checkbox>
            <mat-checkbox formControlName="autonomic" class="example-margin"
              >autonomic</mat-checkbox
            >
          </form>
        </section>
        <section>
          <h5>Lights</h5>
          <form formGroupName="lights">
            <mat-checkbox formControlName="led" class="example-margin">led</mat-checkbox>
            <mat-checkbox formControlName="xenon" class="example-margin">xenon</mat-checkbox>
            <mat-checkbox formControlName="biXenon" class="example-margin">bi-xenon</mat-checkbox>
            <mat-checkbox formControlName="laser" class="example-margin">laser</mat-checkbox>
          </form>
        </section>
        <section>
          <h5>Park assist</h5>
          <form formGroupName="parkingAssist">
            <mat-checkbox formControlName="camera" class="example-margin">camera</mat-checkbox>
            <mat-checkbox formControlName="camera360" class="example-margin"
              >camera 360</mat-checkbox
            >
            <mat-checkbox formControlName="autonomic" class="example-margin"
              >autonomic</mat-checkbox
            >
          </form>
        </section>
      </section>
    </form>
  `,
  styleUrls: ['./vehicle-feature-checkbox-input-form.component.scss']
})
export class VehicleFeatureCheckboxInputFormComponent {
  @Input() equipmentCtrl: FormGroup<ControlsOf<any>> = new FormGroup({});
}
