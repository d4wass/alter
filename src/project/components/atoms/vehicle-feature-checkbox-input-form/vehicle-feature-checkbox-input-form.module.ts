import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ReactiveFormsModule } from '@angular/forms';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { VehicleFeatureCheckboxInputFormComponent } from './vehicle-feature-checkbox-input-form.component';

@NgModule({
  declarations: [VehicleFeatureCheckboxInputFormComponent],
  imports: [CommonModule, ReactiveFormsModule, MatCheckboxModule],
  exports: [VehicleFeatureCheckboxInputFormComponent]
})
export class VehicleFeatureCheckboxInputFormModule {}
