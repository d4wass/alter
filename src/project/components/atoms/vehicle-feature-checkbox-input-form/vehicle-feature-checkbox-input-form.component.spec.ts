import { ReactiveFormsModule } from '@angular/forms';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { VehicleFeatureCheckboxInputFormComponent } from './vehicle-feature-checkbox-input-form.component';

describe('VehicleFeatureCheckboxInputFormComponent', () => {
  let component: VehicleFeatureCheckboxInputFormComponent;
  let spectator: Spectator<VehicleFeatureCheckboxInputFormComponent>;

  const createComponent = createComponentFactory({
    component: VehicleFeatureCheckboxInputFormComponent,
    imports: [ReactiveFormsModule, MatCheckboxModule]
  });

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to snapshot', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
