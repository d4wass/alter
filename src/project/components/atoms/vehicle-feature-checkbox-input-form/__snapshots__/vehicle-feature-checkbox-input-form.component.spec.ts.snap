// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleFeatureCheckboxInputFormComponent should match to snapshot 1`] = `
<app-vehicle-feature-checkbox-input-form>
  <form>
    <section
      class="example-section"
    >
      <mat-checkbox
        class="example-margin"
        formcontrolname="appleCarPlay"
      >
        Apple Car Play
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="androidAuto"
      >
        Android Auto
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="bluetooth"
      >
        Bluetooth
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="usb"
      >
        USB
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="headUpDisplay"
      >
        headup display
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="navigation"
      >
        navigation
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontolname="airConditioning"
      >
        air conditioning
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="keyLess"
      >
        keyless access
      </mat-checkbox>
      <mat-checkbox
        class="example-margin"
        formcontrolname="isofix"
      >
        isofix
      </mat-checkbox>
      <section>
        <h5>
          Cruise Control
        </h5>
        <form
          formgroupname="cruiseControl"
        >
          <mat-checkbox
            class="example-margin"
            formcontrolname="standard"
          >
            standard
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="active"
          >
            active
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="adaptive"
          >
            adaptive
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="autonomic"
          >
            autonomic
          </mat-checkbox>
        </form>
      </section>
      <section>
        <h5>
          Lights
        </h5>
        <form
          formgroupname="lights"
        >
          <mat-checkbox
            class="example-margin"
            formcontrolname="led"
          >
            led
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="xenon"
          >
            xenon
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="biXenon"
          >
            bi-xenon
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="laser"
          >
            laser
          </mat-checkbox>
        </form>
      </section>
      <section>
        <h5>
          Park assist
        </h5>
        <form
          formgroupname="parkingAssist"
        >
          <mat-checkbox
            class="example-margin"
            formcontrolname="camera"
          >
            camera
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="camera360"
          >
            camera 360
          </mat-checkbox>
          <mat-checkbox
            class="example-margin"
            formcontrolname="autonomic"
          >
            autonomic
          </mat-checkbox>
        </form>
      </section>
    </section>
  </form>
</app-vehicle-feature-checkbox-input-form>
`;
